campaign-launcher/interfaceV2/src/components/CampaignsViewDropdown/index.tsx

```diff
@@ -0,0 +1,120 @@
+import { FC, useEffect, useState } from 'react';
+
+import { Box, Button, Menu, MenuItem, Typography } from '@mui/material';
+import { useAccount } from 'wagmi';
+
+import { ArrowDownIcon } from '../../icons';
+import { useWeb3Auth } from '../../providers/Web3AuthProvider';
+import { CampaignsView } from '../../types';
+
+type Props = {
+  campaignsView: CampaignsView;
+  onChange: (view: CampaignsView) => void;
+}
+
+const CampaignsViewDropdown: FC<Props> = ({ campaignsView, onChange }) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+  const { isConnected } = useAccount();
+  const { isAuthenticated } = useWeb3Auth()
+  
+  const open = !!anchorEl;
+
+  useEffect(() => {
+    if ((campaignsView === CampaignsView.JOINED && !isAuthenticated) || (campaignsView === CampaignsView.MY && !isConnected)) {
```

Comment:
This side-effect looks confusing TBH: it's weird that dropdown component accepts a value from the outside, but has to perform extra validations and run side-effects.

It would be better if the list of values was passed from the top, something like:
```
values = [{
 name: 'view name',
 value: 'view const',
 disabled: boolean
}];
```
So the page takes care of overall logic of wha't allowed and what's not and this component is a purely to work with dropdown values.

Otherwise we have a mix of working with these `views`  here and in `Campaigns` component, so it's easy to lose some context/details

Reply:
@dnechay not sure about 'list of values', tho agreed on this effect should not be here

Reply:
@KirillKirill there might be a better way of doing it indeed